src/App.tsx

```diff
@@ -1,24 +1,288 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect, useState, useMemo, useRef,
+} from 'react';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList/TodosList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { Loader } from './components/Loader';
 
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { ERROR_MESSAGES } from './utils/constants/ERROR_MESSAGES';
+import { getFilteredTodos } from './utils/helpers/getFilteredTodos';
+import { SortType } from './types/SortType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodos, setLoadingTodos] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectFilter, setSelectFilter] = useState(SortType.All);
+  const textInputRef = useRef<HTMLInputElement | null>(null);
+  const [isProsessingTodoIds, setIsProsessingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    todoService.getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ERROR_MESSAGES.unableToLoadTodos);
+        // throw new Error(ERROR_MESSAGES.unableToLoadTodos);
+      })
+      .finally(() => {
+        setLoadingTodos(false);
+
+        textInputRef.current?.removeAttribute('disabled');
+        if (textInputRef.current) {
+          textInputRef.current.focus();
+        }
+      });
+  }, []);
+
+  useEffect(() => {
+    let timerError: NodeJS.Timeout | undefined;
+
+    if (errorMessage) {
+      timerError = setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timerError);
+    };
+  }, [errorMessage]);
+
+  useEffect(() => {
+    textInputRef.current?.removeAttribute('disabled');
+    if (textInputRef.current) {
+      textInputRef.current.focus();
+    }
+  }, [todos]);
+
+  const handleAddTodo = (todoTitle: string) => {
+    setErrorMessage('');
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      userId: 0,
+      completed: false,
+    });
+
+    return todoService.addTodo(todoTitle)
+      .then(todo => {
+        setTodos((prevState) => {
+          return [...prevState, todo];
+        });
+      })
+      .catch(() => {
+        setErrorMessage(ERROR_MESSAGES.unableToAddTodo);
+        throw new Error(ERROR_MESSAGES.unableToAddTodo);
+      })
+      .finally(() => {
+        setTempTodo(null);
+
+        textInputRef.current?.removeAttribute('disabled');
+        if (textInputRef.current) {
+          textInputRef.current.focus();
+        }
```

Comment:
You do the same in `useEffect` above

Reply:
@maksym-mishchanchuk If you call some actions several times, you should put them in separate functions and reuse them